pkg/policysync/policy_test.go

```diff
@@ -40,11 +40,11 @@ func TestReflectPolicy(t *testing.T) {
 		t.Fatalf("expected targetRef.Kind to be test, got %s", targetRef.Name)
 	}
 
-	reflectPolicy.UpdateTargetRef(func(targetRef *gatewayapiv1alpha2.PolicyTargetReference) {
```

Comment:
I'd keep the `targetRef` parameter in here. It still compiles and works because the targetRef variable is in the scope from the previous assertion, but the call to `UpdateTargetRef` should be independent to that